packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/Explorer/Dependencies/Dependency/index.tsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import CrossIcon from 'react-icons/lib/md/clear';
+import RefreshIcon from 'react-icons/lib/md/refresh';
+import ArrowDropDown from 'react-icons/lib/md/keyboard-arrow-down';
+import ArrowDropUp from 'react-icons/lib/md/keyboard-arrow-up';
+import algoliasearch from 'algoliasearch/lite';
+import compareVersions from 'compare-versions';
+import Tooltip, {
+  SingletonTooltip,
+} from '@codesandbox/common/lib/components/Tooltip';
+import { formatVersion } from '@codesandbox/common/lib/utils/ci';
+
+import css from '@styled-system/css';
+import {
+  ListAction,
+  Stack,
+  SidebarRow,
+  Select,
+  Text,
+  Link,
+  Button,
+} from '@codesandbox/components';
+
+import { BundleSizes } from './BundleSizes';
+
+interface Props {
+  dependencies: { [dep: string]: string };
+  dependency: string;
+  onRemove: (dep: string) => void;
+  onRefresh: (dep: string, version?: string) => void;
+}
+
+interface State {
+  hovering: boolean;
+  version: null | string;
+  open: boolean;
+  versions: string[];
+}
+
+export class Dependency extends React.PureComponent<Props, State> {
```

Comment:
Is there a way you can make this not a class?

Reply:
can refactor the copied code, didn't make an attempt 👼 8 j